server/src/servershot.js

```diff
@@ -433,17 +435,37 @@ Shot.getRawValue = function(id, deviceId) {
       title: row.title,
       expireTime: row.expire_time,
       deleted: row.deleted,
-      blockType: row.block_type
+      blockType: row.block_type,
+      accountId: row.accountid
     };
   });
 };
 
-Shot.checkOwnership = function(shotId, deviceId) {
+Shot.checkOwnership = function(shotId, deviceId, accountId) {
```

Comment:
I think this can be much simpler, like:

```sql
SELECT data.id
FROM data, devices
WHERE data.id = $1
    AND data.deviceid = device.id
    AND (device.id = $2 OR device.accountid = $3)
```